Sync v2-6-stable with v2-6-test to release 2.6.3rc1

ephraimbuddy opened the pull request:
Time for `2.6.3rc1`!

vijayasarathib committed:
Documentation update on Plugin updates. (#31781)

* Fix for  issue #30796

Tasks forked by the Local Executor are loading stale modules when the modules are also referenced by plugins #30796

(cherry picked from commit 18f2b35c8fe09aaa8d2b28065846d7cf1e85cae2)

vijayasarathib committed:
Increase readability for  prerequisites.rst (#31794)

Fixing minor grammatical errors to increase readability for prerequisites.

(cherry picked from commit 16d9f960944d1ebc0e79fcd17e0c0f63470f2848)

mathbou committed:
Fix airflow module version check when using ExternalPythonOperator and debug logging level (#30367)

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
(cherry picked from commit 406b486823cbd6bd9748c2311cf3fead047a72aa)

dstandish committed:
Add comments explaining need for special "trigger end" log message (#31812)

(cherry picked from commit 2b2b0a736b8f6e091f1362adaa6d484d6034673b)

mis98zb committed:
fix issue: DAG's on_failure_callback is not invoked when task failed during testing dag. (#30965)

---------

Co-authored-by: zengbo <zengbo@microsoft.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 90baac669c446eb4dfb9166d996de59289044983)

Bowrna committed:
adding config changes to docker docs (#31842)

(cherry picked from commit 9cc72bbaec0d7d6041ecd53541a524a2f1e523d0)

potiuk committed:
Add release management command that generates SHOM information (#31835)

We should generate machine-readable SBOM information for our
users to consume, therefore we need to have an automated process
to generate dependency information. We are using cyclonedx standard
to generate the information from two sources:

* constraints of airlfow for PyPI dependencies
* yarn.lock to generate npm dependencies

The script uses officially published docker container of the
CycloneDX organisation that parses both constraints of ours
and yarn lock and produces standard-compliant json output.

The SBOM information is published on our website under
the URL scheme:

https://airflow.apache.org/docs/apache-airflow/<version>/sbom/apache-airflow-sbom-pythonX.Y.json

Since there are different versions for different Python versions,
we are publishing SBOM files for all Python

(cherry picked from commit 72e5a318257d813784f89dab7fa86f1a9111abbb)

stijndehaes committed:
This fixes an issue were a database session got stuck (#31128)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
(cherry picked from commit a6f859eb540bcc168d99d66146a51902502c4217)

vijayasarathib committed:
Increase readability  in start.rst (#31906)

Fixing grammatical error to increase readability

(cherry picked from commit b73eb055b3c3a7772d575b5c159a30e50e3c8451)

potiuk committed:
Improve workflow to release images  (#31916)

* Improve workflow to release images

Adding debugging and explaining that we should use the
stable branch to release the images

* Update dev/README_RELEASE_AIRFLOW.md

(cherry picked from commit d619a98dbfdeb7cc37aabf61987d35eec62457fd)

eladkal committed:
Fix print message in release_candidate_command.py (#31910)

(cherry picked from commit 9a2a5b06957c41f05db974243fcf1577fdd2f0f5)

utkarsharma2 committed:
Fix typos in ISSUE_TRIAGE_PROCESS.rst (#31919)

(cherry picked from commit 6673a404da94228dbb9506d0ab2f51561f82f6c9)

raphaelauv committed:
Bump: kind 0.20.0 (#31922)

(cherry picked from commit 5c1e95deafe9692cd1f3be591e27452fb18f8d5f)

Lee-W committed:
feat(jobs/triggerer_job_runner): add triggerer canceled log (#31757)

Emit log message when trigger is cancelled

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>
Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit a60429eadfffb5fb0f867c220a6cecf628692dcf)

hussein-awala committed:
Fix number param html type in trigger template (#31946)

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit 2d44584f7062113d2db2b8db92230911f8c19e3c)

uranusjr committed:
Fix task.sensor annotation in type stub (#31954)

(cherry picked from commit 7b0fc01a973d508963ef181705ef978c1452c88a)

potiuk committed:
Skip python version check in some pre-commit when loading setup.py (#31965)

Some providers are excluded for some versions of Python (for example
the `apache.hive` provider is excluded on Python 3.11. However
this might make some pre-commits confused when running on Python 3.11
because the list of generated extras will be different for them
than those on other Python versions.

We do not want to set python used in pre-commits to, say, python3.8,
because that would mean that would require our users to have python3.8
as binary available on their system, which is not always setup. Python3
on the other hand is going to be universally available.

In order to mitigate the problem we will run the precommits with
_SKIP_PYTHON_VERSION_CHECK variable set before import of setup is done,
so that the python version check is skipped and all python versions
will produce the same set of extras.

(cherry picked from commit 578cbb24fbd7e4bbeef2e61078c5d333aeda0448)

hussein-awala committed:
Fix masking nested variable fields (#31964)

* Fix masking nested variable fields

Signed-off-by: Hussein Awala <hussein@awala.fr>

* add a unit test

Signed-off-by: Hussein Awala <hussein@awala.fr>

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit e22ce9baed19ddf771db59b7da1d25e240430625)

ephraimbuddy committed:
Fix copy artifacts to svn command (#31975)

This fix uses bash -c to run the svn copy command.
Tested it and it works

(cherry picked from commit d86a1d207c2534643b9031702f98e27e3fc095eb)

eladkal committed:
Update 2.6.2 release date (#31982)

(cherry picked from commit 22b75b2d7fa7e79097539dc3d4b6e316b3e9673f)

potiuk committed:
Add ™ to Airflow in prominent places (#31977)

* Add ™ to Airflow in prominent places

We've started the process of registering the trademark of Airflow and
we need to add ™ marks in prominent places of our website.

* Update docs/apache-airflow/index.rst

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
(cherry picked from commit 10aa704e3d87ce951cb79f28492eed916bc18fe3)

hussein-awala committed:
Add a check for trailing slash in webserver base_url (#31833)

* Remove right trailing / from webserver base_url

Signed-off-by: Hussein Awala <hussein@awala.fr>

* use url join instead of removing trailing slash

Signed-off-by: Hussein Awala <hussein@awala.fr>

* raise an exception when base_url contains a trailing slash

Signed-off-by: Hussein Awala <hussein@awala.fr>

* Update airflow/www/extensions/init_wsgi_middlewares.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit fe4a6c843acd97c776d5890116bfa85356a54eee)

potiuk committed:
Add command to generate requirements that are needed by a provider (#31943)

This new SBOM command will generate the requirement file for the
specified provider. It will exclude all the requirements that Airflow
adds on its own. It will be used in the future to generate per
provider sbom.

The SBOM commands have also been separated out to a new command.

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit d2c038e0437a23ce1f41960ce6e27216d9f9e579)

hussein-awala committed:
Fix number param html type in trigger template (#31980)

* Keep valuetype="number" for both param number and integer types and change only type value

Signed-off-by: Hussein Awala <hussein@awala.fr>

* Update airflow/www/templates/airflow/trigger.html

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 232771869030d708c57f840aea735b18bd4bffb2)

hussein-awala committed:
fix connection uri parsing when the host includes a scheme (#31465)

* update _parse_from_uri and get_uri  methods, and add tests for connection model

* some fixes from review

(cherry picked from commit 0560881f0eaef9c583b11e937bf1f79d13e5ac7c)

o-nikolas committed:
Check for DAG ID in query param from url as well as kwargs (#32014)

Previously the dag id was only being checked in request args and form
but not kwargs, so it was possible for the id when passed as kwargs
to be None. This can allow auth for a user who does not have the
permissions to view a particular DAG.

(cherry picked from commit ac65b82eeeeaa670e09a83c7da65cbac7e89f8db)

Lee-W committed:
Catch the exception that triggerer initialization failed (#31999)

* fix(jobs/triggerer_job_runner): catch the exception that triggerer initialization failed

* test(jobs/triggerer_job_runner): add test case for checking TriggerRunner.update_trigger catch triggerer change exception

* refactor(jobs/triggerer_job_runner): catch multiple exception cases in the same try catch block when initialing trigger

* refactor(jobs/triggerer_job_runner): remove unnecessary TypeError catching

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* Revert "refactor(jobs/triggerer_job_runner): remove unnecessary TypeError catching"

This reverts commit ac561a2416f16a8f9a3b211715fd4a7e012b47ad.

* Revert "refactor(jobs/triggerer_job_runner): catch multiple exception cases in the same try catch block when initialing trigger"

This reverts commit 4f3b24b8a7698973fe319618b6ff9f459dc0155a.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit a1ba15570219d2fe77466367e84fafa92cbdb24e)

potiuk committed:
Less precise check for error message in trigger initialization (#32035)

The test for error message (introduced in #31999) in trigger
initialization expected precise error message, however Python 3.10
changed the details printed when TypeError was converted to string
(including the information about class being created not only method).
This caused the tests to fail on Python 3.10 and 3.11.

Making two asserts with two parts of the message expected solves
the problem for all python versions.

(cherry picked from commit 12c8099846a55ba43bf7d48bab3d01ab697f97a4)

dstandish committed:
Fix hashing of dag_dependencies in serialized dag (#32037)

(cherry picked from commit c76d57a67f1091e49cc9a8255054970e05ab1de5)

hussein-awala committed:
Fix ExternalTaskSensor when there is not task group TIs for the current execution date (#32009)

* Add a check on none TIs for the current execution date

Signed-off-by: Hussein Awala <hussein@awala.fr>

* replace inline if-else by old one

Signed-off-by: Hussein Awala <hussein@awala.fr>

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit 14eb1d3116ecef15be7be9a8f9d08757e74f981c)

ephraimbuddy committed:
Return None if an XComArg fails to resolve in a multiple_outputs Task (#32027)

* Return None if an XComArg fails to resolve in a multiple_outputs Task

Tasks with multiple_outputs set to True returns XComs with different keys
which are not known to Airflow. Because they have multiple_outputs set, we
should return None if we can't find the XCom, just like we return None when
the key is equal to XCOM_RETURN_KEY known to Airflow.

Closes: https://github.com/apache/airflow/issues/29199

* Apply suggestions from code review

(cherry picked from commit 79eac7687cf7c6bcaa4df2b8735efaad79a7fee2)

potiuk committed:
Add pkgconf to system dependencies for the CI/prod image (#32075)

The new mysqlclient 2.2.0 requires pkgconf to be built. This change
adds it to both CI image and build segment of the PROD image.

(cherry picked from commit 2892fe89fa71fb7d6d16427dd344764d4bccaa29)

pierrejeambrun committed:
Use re2 for matching untrusted regex (#32060)

(cherry picked from commit 116e607ddcb32480e57c342f48226545ac6fc315)

potiuk committed:
Remove casting of forward_credentials for shell and start-airflow breeze (#32087)

The forward-credentials flag was supposed to be disabled when you run
start-airflow and shell commands. However the flag boolean value has
been casted to str so False became "False" and forward-credentials
for these two commaands have been enabled by default no matter if
the flag was used. In effect for example .docker configuration
and credentials have been forwarded to inside the docker container
via `.docker` folder mapped, and in case you had some credentials
helper or plugin configured there, it was also set for docker
configuration.

Using the boolean value fixes the problem. Now you shoudl specify
the --forward-credentials flag explicitly to forward credentials.

(cherry picked from commit 3ead4aa610b3381c8b3f4eecc2be34bb8c9ba16b)

pankajkoti committed:
Update comment for editable installs wrt setuptools (#32090)

(cherry picked from commit ca13c7b77ea0e7d37bfe893871bab565d26884d0)

vijayasarathib committed:
Adding missing hyperlink to the tutorial documentation (#32105)

* Adding missing hyperlink to the tutorial documentation

(cherry picked from commit 4a6efe520f6b357002072bf5955eb731d5bfd716)

potiuk committed:
Exit asset compilation fork in start-airflow after it completed (#32114)

The `start-airflow` command uses fork to start parallell asset
compilation in the background so that it can happen while docker
compose initializes. Unfortunately this fork child did not have
sys.exit() so it returned from the function and continued to run
second docker-compose in the background. In case asset compilation
was not needed this could happen in parallel and both processes
attempted to start two docker-compose commands in parallel.

This was not visible in "dev" mode - because asset compilation
never completed there also - when asset compilation was needed,
it took some time before it completed, and the effect of it were
not visible, because the forked process did not get terminal
output (it has been taken over by tmux by the time it started to
use it) and could not grab forwarded ports, so it was running but
largely invisible.

However when asset compilation was not needed, the two processes
started to do the same things at the same time - so a lot of
the output has been duplicated and for example the line output
has been broken because the same messages were overwriting over
each other and canceling the effect of EOL printed to terminal.

With this change, the forked process exits as soon as the
asset compilation is completed and does not repeat the same
steps that the parent process is doiing.

(cherry picked from commit 5dddf579907d0eaec28a17cf62e32da9a8d627ca)

rohan472000 committed:
added difference between Deferrable and Non-Deferrable Operators (#31840)

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 0db0ff14da449dc3dbfe9577ccdb12db946b9647)

potiuk committed:
Avoid permission error when asset compilation exists before killing it (#32116)

After #32114 the atexit registered killpg produces Permission Error in
stdout when the group was missing (basically when the asset compilation
stopped).

Changing it to ignore the error avoids the false negative appear in the
output.

(cherry picked from commit 3aff742d3941ddcca60197e0327c46bd39471391)

potiuk committed:
Add release management command to retreive provider version metadata (#32119)

This new command will refresh and store generated provider version
metadata - including the time when provider has been released and
which airflow version it has been associated with.

This information will eventually be used in generating provider's
changelogs, and it will be used to generate provider-specific
dependencies (we need to know which airflow version provider
specific dependencies should be compared with).

This is the next step in publishing SBOM information, but it will
also be accompanied by updating more information for past releases
of the providers.

(cherry picked from commit 63e11171b833ec87ebdc169377b18406bcf76abe)

hussein-awala committed:
Flash an error msg instead of failure in `rendered-templates` when map index is not found (#32011)

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit 62a534dbc7fa8ddb4c249ade85c558b64d1630dd)

Lee-W committed:
docs(authoring-and-scheduling/deferring): add a note that we'll need to restart triggerer to reflect any trigger change (#32140)

(cherry picked from commit 51dbbaf40b90aa676cc706fd8c40c12cac66b9cc)

potiuk committed:
Add airflow to docs packages if only docs/apache-airflow/* files changed (#32149)

Selective checks for docs building missed the case where airflow
package docs should also be build when docs/appache-airflow/*
files changed.

(cherry picked from commit de585f521b5898ba7687072a7717fd3b67fa8c5c)

Commit:
Update callbacks.rst (#32148)

(cherry picked from commit 45bd9c993ee87789e01aae55f8abb34a0ff1de48)

vijayasarathib committed:
Documentation Changes (#32185)

Readability change in the kubernetes.rst tutorial file

(cherry picked from commit 23b8e839a35e84d57c5cf38b0b21171ac3bd1ecc)

potiuk committed:
Add libeos_dev on ARM runtime PROD image (#32200)

This is a follow-up after #31437. Seems that for ARM image, the
BigQuery libraries require libgeos-dev library installed on
system level not only to be built but also to be imported.

(cherry picked from commit 771362af4784f3d913d6c3d3b44c78269280a96e)

tomrutter committed:
Ensure that main triggerer thread exits if the async thread fails (#32092)

Co-authored-by: tom.rutter <tom.rutter@zurich.com>
(cherry picked from commit e585b588fc49b1b1c73a8952e9b257d7a9e13314)

potiuk committed:
Remove executor_class from Job - fixing backfil for custom executors (#32219)

The executor_class in job was used in the past to pass information
about executor being used but it has been replaced by executor
passed via JobRunner when needed. The executor_class has only
been used in backfill job, but it has been used wrongly - because
it had no fully qualified path there.

The right way of using it is to check the executor passed via
JobRunner, and since executor_class is not used any more, we
can safely remove it.

This fixes a bug when custom exceutors could not be used
during backfill.

Co-authored-by: Andrew Halpern <57367683+adh-wonolo@users.noreply.github.com>
(cherry picked from commit df4026a47aa6141142e186ec853ba286c2be0e95)

mobuchowski committed:
add mobuchowski to committers breeze variable (#32228)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>
(cherry picked from commit b6ec05187e14f7ca9814b4ce90755f538a1a7d7a)

phanikumv committed:
Add phanikumv to committers (#32229)

(cherry picked from commit fb54e1fcaa046885b14b01edea9342631908d051)

tirkarthi committed:
Use mapIndex to display extra links per mapped task. (#32154)

(cherry picked from commit 98c47f48e1b292d535d39cc3349660aa736d76cd)

vincbeck committed:
Add vincbeck as committer (#32240)

(cherry picked from commit c947533dba02b443595068f403429e233a648faa)

uranusjr committed:
Add comment to warn off a potential wrong fix (#32230)

(cherry picked from commit 73dd48083d1102b81b76ff6f5668cc41261c362e)

fritz-astronomer committed:
Add links to DAGRun / DAG / Task in templates-ref.rst (#32245)

* Update templates-ref.rst

Add links to DAGRun / DAG / Task

* add TaskInstance link

(cherry picked from commit b6477d0c3d498a8a1962fa67e0a92326ee6393fb)

potiuk committed:
Fix behaviour of LazyDictWithCache wheni import fails (#32248)

When #15330 added docker.task, it also optimized replacement of the
callable with it's result in LazyDictWithCache. LazyDictWithCache
is used by Provider's Manager to optimize access to hooks - basically
hook is only actually imported, when it is accessed. This helps with
speeding up importing of connection information

The optimization added result of running callable to _resolved
set, but it missed the case when None was returned. Previously,
when None was returned, the callable was not replaced and it
was called again. After the change - the _resolved set was
updated with the key and None was returned. But since the key
has not been replaced, next time when the same key was retrieved,
the original "callable" was returned, not the None value. So if
callable returned None, and the same key was retrieved twice, the second
time, instead of None, the dictionary returned Callable.

This PR fixes it by setting the value to dictionary even if it
was None.

(cherry picked from commit 462df0d05abbf023ea367ad2591cc889a49c815a)

mobuchowski committed:
fix bugged singleton implementation (#32218)

* fix bugged singleton implementation

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>
(cherry picked from commit e2e707ca2720bf066228a136c36c17fd6eda802c)

utkarsharma2 committed:
Decode old-style nested Xcom value (#31866)

The deserializer was not properly dealing with nested and wrapped old-style xcom values.
---------

Co-authored-by: bolkedebruin <bolkedebruin@users.noreply.github.com>
(cherry picked from commit bd32467ede1a5a197e09456803f7cebaee9f9b77)

potiuk committed:
Update git-sync description in Helm Chart documentation (#32181)

There are quite a few recurring themes when it comes to using
git-sync for DAG synchronisation and this documentation is an
attempt to capture results of a number of discussions and
conversations. It adds some notes that might make it possible to
make more informed decisions by our users and Deployment managers
who want to make decisions on how they should synchronize their
DAGs.

The changes include:

* notes on potential side-effects one has to be aware when
  using both git-sync and persistence together (there are some
  unobvious operations performed by git-sync that might affect
  performances of persistence solutions)

* notes on how you can use multiple git repositories with git-sync
  using submodule approach - including link to a real-life use
  case from Airflow summit where it has been used in production
  for 100s of repositories.

(cherry picked from commit b6ca28e55026acb98fb50aa098c7f621a165cd62)

potiuk committed:
Separate out advanced logging configuration (#32131)

The "advanced logging configuration" applies not only to task
logs but also to component logs and you can use it not only to
configure custom way how task logs are created but also custom
way how "regular" component logs are created.

This has been a source of confusion for those who wanted
to configure (for example) elasticsearch or opensearch for the
whole airflow deployment, because the "advanced configuration"
and how to modify standard configuration chapter was a small
section in "task logging".

This change extracts "advanced logging configuration" to separate
page right under the "logging and monitoring" and directs the
user from the "task" logging section to this page. It also
adds a bit more explanation on how standard Python logging
framework is leveraged here and links to Python logging documentation
for those who never used it before, to understand more about
Loggers, Handlers and Formatters.

Co-authored-by: Akash Sharma <35839624+Adaverse@users.noreply.github.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit ead2530d3500dd27df54383a0802b6c94828c359)

potiuk committed:
Update references to restructured documentation from airflow core (#32282)

The change #32131 restructured some of the code for documentation
and it broke references to apache-airlfow from "providers" doc
package - it has not been visible in selective PR (not sure the reason)
but this PR fixes it.

(cherry picked from commit ef49e595f1ab729cdb72198a3ee47c7dcd182da2)

potiuk committed:
Add apache-airflow-providers to filter in selective checks if needed (#32284)

When provider docs in "docs/apache-airflow-providers" changed, and
selective docs build was used, the "apache-airflow-providers" package
was not added to the filter - thus errors in those docs were not
checked. This caused  #32131 to fail the main build (fixed in
the #32282). Added logic to handle this case.

(cherry picked from commit ba05a1e8706abffe057fa83d891b44125d5913e6)

hussein-awala committed:
Fix triggerers alive check and add a new conf for triggerer heartbeat rate (#32123)

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit d117728cd6f337266bebcf4916325d5de815fe03)

potiuk committed:
Fix Pydantic 2 pickiness about model definition (#32307)

The new Pydantic 2 has been released on 30th of June and it is a
bit more picky about model definition, thus causing Airflow to
fail because there were wrong definitions of models (previously
corretly validated).

Fixes: #32301
(cherry picked from commit 4d84e304b86c97d0437fddbc6b6757b5201eefcc)

potiuk committed:
Add information for users who ask for requirements (#32262)

* Add information for users who ask for requirements

This change is based on a number of discussions with the users asking
what are the minimum requirements for Airflow to run.

While we cannot give precise answer, we should also make the users
aware that simple answers are not possible, and that when they are
deciding to install airflow and manage it on their own, they also
take the responsibility to monitor and adjust the resources they
need based on the monitoring they have to run.

* Apply suggestions from code review

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

* Update docs/apache-airflow/installation/index.rst

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
(cherry picked from commit f6db66e16374e504665972feba0831d4148c6d50)

potiuk committed:
Update security model of Airflow (#32098)

* Update security model of Airflow

This change updates the security model of Airflow to better explain
what are the capabilities of various kinds of users in Airflow
deployments and give both users and security researchers a way to
understand what security measures they can take and whether they
can qualify potential security issues in Airflow properly - taking
into account that various users of Airflow have various capabilities
and behaviours considered by some of the users as security
vulnerabilities, are standard capabilities of the users.

It also splits the security information of ours in two separate
pages:

* .github/SECURITY.md where we explain how to report the issues
  to Apache Airflow security team by the researchers

* documentation security/index.html which is available via
  Airflow Website where we explain what our security model is
  and the different kinds of users we have.

Both serve slightly different purpose and both contain cross-reference
links to each other in order to be able to redirect people who read
about the security model to find out how they can report the issues
but also to guide security researchers who want to assess whether
their findings are real vulnerabilities, or rather normal behaviours
following the Airflow Security model.

Security has been also moved to be a top level topic, so that it
is easier to find and navigate to. Old links have been redirected
to the new locations.

Also chapters were added explaining Airflow vs. Providers security
releases, what is the relation between Airflow and Providers
security issues and how users should treat security announcements
in providers.

* Update .github/SECURITY.md

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

* Apply Niko's suggestions from code review

Co-authored-by: Niko Oliveira <onikolas@amazon.com>

* fixup! Apply Niko's suggestions from code review

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
Co-authored-by: Niko Oliveira <onikolas@amazon.com>
(cherry picked from commit 4efbcdc8fc48e929f18fbe75c61a5125072f71fc)

potiuk committed:
Limit Pydantic to < 2.0.0 until we solve 2.0.0 incompatibilities (#32312)

The Pydantic 2.0.0 release on 30th of June broke a number of CI
tests (including some K8S tests).

We limit it temporarily until the incompatibilities are fixed.

Related: #32311
(cherry picked from commit 2811ba710991d71466afa21c500ee55aeda5d8a1)

ephraimbuddy committed:
Sanitize `DagRun.run_id` and allow flexibility (#32293)

This commit sanitizes the DagRun.run_id parameter by introducing a configurable option.
Users now have the ability to select a specific run_id pattern for their runs,
ensuring stricter control over the values used. This update does not impact the default run_id
generation performed by the scheduler for scheduled DAG runs or for Dag runs triggered without
modifying the run_id parameter in the run configuration page.
The configuration flexibility empowers users to align the run_id pattern with their specific requirements.

(cherry picked from commit 05bd90f563649f2e9c8f0c85cf5838315a665a02)

potiuk committed:
Fix SemVer link in security documentation (#32320)

The SemVer link was an md-style one, we should change it rst-style

pierrejeambrun committed:
Fix ts/eslint warnings (#32321)

(cherry picked from commit b6eeccdfb67409f5035adc3ab8af97c4c03a40dc)

cvaliente committed:
Render list items in rendered fields view (#32042)

---------

Co-authored-by: clemens.valiente <clemens.valiente@grabtaxi.com>
(cherry picked from commit 2b5d4e319b82267349aa20cb3230f3b2ad35adce)

vijayasarathib committed:
Clarify Listener API behavior  (#32269)

* Fixes: #31180 - Plugin for listeners - on_dag_run_running hook ignored

Documentation update for Plugin for listeners - on_dag_run_running hook ignored
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

(cherry picked from commit ab2c861dd8a96f22b0fda692368ce9b103175322)

pankajkoti committed:
Update index.rst doc to correct grammar (#32315)

* Update index.rst doc to correct grammar

* Address @uranusjr's comments

(cherry picked from commit 43f3e57bf162293b92154f16a8ce33e6922fbf4e)

Adaverse committed:
Fix try_number shown in the `task?task_id=&dag_id=&execution_date=` (#32361)

* try_number fix

* change test name

(cherry picked from commit a8e4b8aee602e8c672ab879b7392a65b5c2bb34e)

pragnareddye committed:
Update python.rst with multi-Scheduler LocalExecutor info (#32310)

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
(cherry picked from commit e53320d62030a53c6ffe896434bcf0fc85803f31)

potiuk committed:
Retry transactions on occasional deadlocks for rendered fields (#32341)

This is a follow-up on #18616 where we introduced retries on the
occassional deadlocks when rendered task fields have been deleted
by parallel threads (this is not a real deadlock, it's because
MySQL locks too many things when queries are executed and will
deadlock when one of those queries wait too much).

Adding retry - while not perfect - should allow to handle the
problem and significantly decrease the likelihood of such
deadlocks.

We can probably think about different approach for rendered fields,
but for now retrying is - I think - acceptable short-term fix.

Fixes: #32294
Fixes: #29687
(cherry picked from commit c8a3c112a7bae345d37bb8b90d68c8d6ff2ef8fc)

holmen1 committed:
Update pipeline.rst (#32369)

Typo in final dag, merge_data.

(cherry picked from commit 383efbcbe00a0ce25251ba3a06a625f0ae40d806)

potiuk committed:
Hide sensitive values from extra in connection edit form (#32309)

The fields that are sensitive (i.e password field is used
to show them in Connection edit view) should also be hidden
when they are stored as "extra" in the form extra field.

This PR handles both - replacing such values with a
placeholder as well as not updating the value if the
placeholder has not been modified.

(cherry picked from commit d01248382fe45a5f5a7fdeed4082a80c5f814ad8)

hussein-awala committed:
Fix `operator_extra_links` property serialization in mapped tasks (#31904)

* reproduce the problem in a unit test

* Fix operator_extra_links serialization

* replace fget by __get__

(cherry picked from commit 3318212482c6e11ac5c2e2828f7e467bca5b7245)

eladkal committed:
Fix failing main due to new release of jsonschema (#32405)

(cherry picked from commit 89d5e8b456c3f8e2cbe2691a4e1fa52ef2c1b6c5)

potiuk committed:
Add information on ARM MySQL support in Docker images (#32401)

The change #29519 added experimental MariaDB client support for
MySQL for ARM images but we missed that in the changelog.

(cherry picked from commit ef75a3a6757a033586c933f7b62ab86f846af754)

pierrejeambrun committed:
Use linear time regular expressions (#32303)

The standard regexp library can consume > O(n) in certain circumstances.
The re2 library does not have this issue.

(cherry picked from commit ee38382efa54565c4b389eaeb536f0d45e12d498)

ephraimbuddy committed:
Update version to 2.6.3

ephraimbuddy committed:
Update RELEASE_NOTES.rst

potiuk reviewed (commented):
Reviewed the commits. LGTM.

ephraimbuddy committed:
Update trigger url to fix test

potiuk committed:
Limit click until we fix mypy issues (#32413)

Click 8.1.4 breaks our mypy checks. The upper limit can be lifted when
the #32412 issue is resolved

(cherry picked from commit bb97bf21fd320c593b77246590d4f8d2b0369c24)

xuganyu96 committed:
Use WebEncoder to encode DagRun.conf in DagRun's list view (#32385)

* Use WebEncoder to encode DagRun.conf in DagRun's list view

* test case for json_f with WebEncoder

* Always use WebEncoder in web views

(cherry picked from commit 4bbb633830f8380d7776d12ddd3557dcaaa3133f)

eladkal committed:
Add explanation about tag creation and cleanup (#32406)

* Add explanation about tag creation and cleanup

(cherry picked from commit b1c1f97744e735c5230fc658cf8deb71aa770f9b)

potiuk committed:
fixup! feat(jobs/triggerer_job_runner): add triggerer canceled log (#31757)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Hide sensitive values from extra in connection edit form (#32309)
+- Sanitize ``DagRun.run_id`` and allow flexibility (#32293)
+- Add triggerer canceled log (#31757)
+- Fix try_number shown in the ``task?task_id=&dag_id=&execution_date=`` (#32361)
```
```suggestion
- Fix try number shown in the task view (#32361)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Return ``None`` if an XComArg fails to resolve in a multiple_outputs Task (#32027)
+- Check for DAG ID in query param from url as well as kwargs (#32014)
+- Flash an error message instead of failure in ``rendered-templates`` when map index is not found (#32011)
+- Fix ``ExternalTaskSensor`` when there is not task group TIs for the current execution date (#32009)
```
```suggestion
- Fix ``ExternalTaskSensor`` when there is no task group TIs for the current execution date (#32009)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Decode old-style nested Xcom value (#31866)
+- Add a check for trailing slash in webserver base_url (#31833)
+- Fix connection uri parsing when the host includes a scheme (#31465)
+- This fixes an issue were a database session got stuck (#31128)
```
```suggestion
- Fix database session closing with ``xcom_pull`` and ``inlets`` (#31128)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+""""""""""""""""
+- Add explanation about tag creation and cleanup (#32406)
+- Minor updates to docs(#32369,#32315,#32310)
+- Clarify Listener API behavior  (#32269)
```
```suggestion
- Clarify Listener API behavior (#32269)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+Doc only changes
+""""""""""""""""
+- Add explanation about tag creation and cleanup (#32406)
+- Minor updates to docs(#32369,#32315,#32310)
```
```suggestion
- Minor updates to docs (#32369,#32315,#32310)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Minor updates to docs(#32369,#32315,#32310)
+- Clarify Listener API behavior  (#32269)
+- Add information for users who ask for requirements (#32262)
+- Add links to DAGRun / DAG / Task in templates-ref.rst (#32245)
```
```suggestion
- Add links to DAGRun / DAG / Task in Templates Reference (#32245)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Adding missing hyperlink to the tutorial documentation (#32105)
+- Added difference between Deferrable and Non-Deferrable Operators (#31840)
+- Add comments explaining need for special "trigger end" log message (#31812)
+- Increase readability for  prerequisites.rst (#31794)
```
This one can probably be skipped, or included in the "minor updates" catchall above.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,73 @@
+- Documentation update on Plugin updates. (#31781)
+- Fix SemVer link in security documentation (#32320)
+- Update security model of Airflow (#32098)
+- Update references to restructured documentation from airflow core (#32282)
```
```suggestion
- Update references to restructured documentation from Airflow core (#32282)
```
